### PR TITLE
Add environment variables for telemetry

### PR DESCRIPTION
**Pull request by GeekArthur**
This PR adds the environment variables to devfile community registry

Signed-off-by: Jingfu Wang <jingfwan@redhat.com>

### CODE REVIEW

File: .ci/deploy/devfile-registry.yaml [hunk 1]
```diff
@@ -69,6 +69,11 @@ objects:
             limits:
               cpu: 100m
               memory: ${DEVFILE_INDEX_MEMORY_LIMIT}
+          env:
+          - name: ENABLE_TELEMETRY
+            value: true
```
Comment (johnmcollier) [inline .ci/deploy/devfile-registry.yaml:74]: This field should be be configurable with a parameter under https://github.com/devfile/registry/blob/b05971c7cf9fc52f280dc23690d25bc01511b479/.ci/deploy/devfile-registry.yaml#L156

And the default value should be false.

File: .ci/deploy/devfile-registry.yaml [hunk 2]
```diff
@@ -69,6 +69,11 @@ objects:
...
+          - name: REGISTRY
```
Comment (johnmcollier) [inline .ci/deploy/devfile-registry.yaml:75]: What's this environment variable for exactly? I thought we were only going to use one project in segment?

And we should use a more descriptive environment variable name.

**Review by johnmcollier — CHANGES_REQUESTED** [review]

File: .ci/deploy/devfile-registry.yaml (hunk 2 above)
Comment (johnmcollier) [reply .ci/deploy/devfile-registry.yaml:75]: It should also be configurable with a parameter

Comment (GeekArthur) [reply .ci/deploy/devfile-registry.yaml:75]: Yes, only one project for segment and woopra, this is the registry name that is used as identifier for telemetry to know  which registry the data come from. I will update the description.

File: .ci/deploy/devfile-registry.yaml (hunk 1 above)
Comment (johnmcollier) [reply .ci/deploy/devfile-registry.yaml:74]: The parameter will be set in AppSRE's Gitlab instance, where we have the deployment configurations for staging and prod

Comment (GeekArthur) [reply .ci/deploy/devfile-registry.yaml]: Got it, thanks

**Review by johnmcollier — CHANGES_REQUESTED** [review]
Thanks for making the changes Jingfu. Two more changes needed:
1) Applying the template gives the following error:
```
johncollier@jcollier-mac hosted-registry % oc process -f devfile-registry.yaml | oc apply -f - 
error: failed to read input object (not a Template?): unable to decode "devfile-registry.yaml": v1.Template.Parameters: []v1.Parameter: v1.Parameter.Value: ReadString: expects " or n, but found f, error found in #10 byte of ...|,"value":false},{"de|..., bigger context ...|ry enablement","name":"ENABLE_TELEMETRY","value":false},{"description":"The registry name that is us|...
```
The `env` field in Kubernetes expects strings for values, so I think `ENABLE_TELEMETRY` being a bool is throwing it off. Surrounding the `false` in double quotes fixed it for me.

2) Can you update the `.ci/openshift_integration.sh` script in this repo to point to this template? It looks like this repo's tests are still using the old template in registry-support (that I should've removed)

**Comment by GeekArthur** [comment]
> Thanks for making the changes Jingfu. Two more changes needed:
> 
>     1. Applying the template gives the following error:
> 
> 
> ```
> johncollier@jcollier-mac hosted-registry % oc process -f devfile-registry.yaml | oc apply -f - 
> error: failed to read input object (not a Template?): unable to decode "devfile-registry.yaml": v1.Template.Parameters: []v1.Parameter: v1.Parameter.Value: ReadString: expects " or n, but found f, error found in #10 byte of ...|,"value":false},{"de|..., bigger context ...|ry enablement","name":"ENABLE_TELEMETRY","value":false},{"description":"The registry name that is us|...
> ```
> 
> The `env` field in Kubernetes expects strings for values, so I think `ENABLE_TELEMETRY` being a bool is throwing it off. Surrounding the `false` in double quotes fixed it for me.
> 
>     1. Can you update the `.ci/openshift_integration.sh` script in this repo to point to this template? It looks like this repo's tests are still using the old template in registry-support (that I should've removed)

For the second change, I assume we trigger the integration test script from this repo so I use relative path `deploy/devfile-registry.yaml` to point to the integration test script, also I notice we don't have route.yaml in this repo so it still use the route.yaml from registry-support repo, another alternative is that copying over route.yaml to this repo.

**Comment by johnmcollier** [comment]
Yeah, use deploy/devfile-registry.yaml to refer to it.

And yeah, feel free to copy over the route.yaml as well.

**Comment by GeekArthur** [comment]
@johnmcollier Comments addressed, please review again.

**Comment by johnmcollier** [comment]
@GeekArthur Sorry Jingfu, looks like I was wrong about the paths. I forgot that the OpenShift CI executes in the root of the repository, so it needs to be `.ci/deploy` for the registry and the route yamls.

**Comment by GeekArthur** [comment]
> @GeekArthur Sorry Jingfu, looks like I was wrong about the paths. I forgot that the OpenShift CI executes in the root of the repository, so it needs to be `.ci/deploy` for the registry and the route yamls.
> 
> Otherwise it looks good!

@johnmcollier Updated, please review again

**Review by johnmcollier — APPROVED** [review]

**Pull request merged**